### PR TITLE
allow query by date

### PR DESCRIPTION
**Pull request by adamlofting**
I haven't got the DB setup to test this, but thought adding in this much might save you some time :)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -12,6 +12,25 @@ var async = require("async");
 var http = require("http");
 var server = http.createServer();
 server.on("request", function(req, res) {
+
+  var date;
+  if (req.query.date) {
```
Comment (jbuck) [inline server.js]: This is just plan [http](http://nodejs.org/api/http.html#http_message_url) so you'll need to parse the querystring manually like so:

``` javascript
var querystring = require('querystring');
var date = querystring.parse(req.url);
```

File: server.js [hunk 2]
```diff
@@ -12,6 +12,25 @@ var async = require("async");
...
+    date = new Date(req.query.date);
+    if ( Object.prototype.toString.call(date) === "[object Date]" ) {
```
Comment (jbuck) [inline server.js]: `date` will always be a Date object, so you can remove this if

File: server.js [hunk 3]
```diff
@@ -12,6 +12,25 @@ var async = require("async");
...
+      if ( isNaN( date.getTime() ) ) {
+        date = null;// date is not valid
+      }
+    }
+    else {
```
Comment (jbuck) [inline server.js]: this branch can be removed

File: server.js [hunk 4]
```diff
@@ -12,6 +12,25 @@ var async = require("async");
 var http = require("http");
 var server = http.createServer();
 server.on("request", function(req, res) {
+
+  var date;
+  if (req.query.date) {
+    date = new Date(req.query.date);
+    if ( Object.prototype.toString.call(date) === "[object Date]" ) {
+      if ( isNaN( date.getTime() ) ) {
```
Comment (jbuck) [inline server.js]: I'd move this check down to line 29

File: server.js [hunk 5]
```diff
@@ -35,9 +54,9 @@ server.on("request", function(req, res) {
       return res.end(JSON.stringify({error: err.toString()}));
     }
 
-    var one_year_ago = Date.now() - (365 * 24 * 60 * 60 * 1000);
+    var one_year_ago = date - (365 * 24 * 60 * 60 * 1000);
```
Comment (jbuck) [inline server.js]: You'll need to convert `date` into an integer by calling `date.valueOf()` for this comparison to work

File: server.js [hunk 6]
```diff
@@ -35,9 +54,9 @@ server.on("request", function(req, res) {
...
     var total_active_contributors = 0;
-    var seven_days_ago = Date.now() - (7 * 24 * 60 * 60 * 1000);
+    var seven_days_ago = date - (7 * 24 * 60 * 60 * 1000);
```
Comment (jbuck) [inline server.js]: valueOf() here too!

**Comment by jbuck** [comment]
I think the comparison might be broken too on line 71 and 74. I think you'll need to add an additional comparator like `&& latestContribution < date`

File: server.js (hunk 2 above)
Comment (adamlofting) [inline server.js]: I think it's possible that `date` is not a Date if it's passed an invalid date string via the URL: http://stackoverflow.com/questions/1353684/detecting-an-invalid-date-date-instance-in-javascript

**Comment by jbuck** [comment]
Or maybe not? I'm not sure now... would need a whiteboard to figure it out. I feel like the ranges could work though!

**Comment by cadecairos** [comment]
https://github.com/jbuck/park-warden/commit/e06b965256d69dcf62c54f5aa6ae7918dfa34033

**Pull request closed without merging**